packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -91,22 +96,21 @@ class MemoryController extends DisposableController
       case DevToolsMode.offlineData:
         assert(connectedDiff == null && connectedProfile == null);
         await maybeLoadOfflineData(
-          PerformanceScreen.id,
+          ScreenMetaData.memory.id,
           createData: (json) => OfflineMemoryData.parse(json),
-          shouldLoad: (data) => !data.isEmpty,
+          shouldLoad: (data) => true,
         );
-        // If shouldLoad returns false, previous line is noop, so data should be initialized.
-        if (!isInitialized.value) _initializeData();
+        assert(_initialized.isCompleted);
```

Comment:
this will not be completed if maybeLoadOfflineData was a noop

Reply:
I don't think it would if the wrapped if check returns false `offlineDataController.shouldLoadOfflineData(screenId)`

Reply:
Fixed.